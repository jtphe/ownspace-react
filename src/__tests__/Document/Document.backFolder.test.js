import React from 'react';
import BackFolder from '@components/Document/BackFolder';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import * as Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { View } from 'react-native';
import { CLIENT_COLOR_PRIMARY } from '@constants';

describe('BackFolder component tests', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<BackFolder visible={true} />);
  });
  it('BackFolder snapshot', async () => {
    let component;
    await renderer.act(async () => {
      component = renderer.create(<BackFolder visible={true} />);
    });
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Icon back', () => {
    const iconBack = wrapper.find(View).get(0).props.children;
    expect(iconBack[0].props.name).toEqual('arrow-left');
    expect(iconBack[0].props.size).toEqual(18);
    expect(iconBack[0].props.color).toEqual(CLIENT_COLOR_PRIMARY);
    expect(iconBack[0].props.allowFontScaling).toEqual(false);
  });
});

Enzyme.configure({ adapter: new Adapter() });
