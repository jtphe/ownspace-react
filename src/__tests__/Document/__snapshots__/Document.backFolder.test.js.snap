// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackFolder component tests BackFolder snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "margin": 15,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#003466",
          "fontSize": 18,
        },
        undefined,
        Object {
          "fontFamily": "Feather",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ÔÑê
  </Text>
  <View
    style={
      Object {
        "marginLeft": 7,
      }
    }
  >
    <Text
      ellipsizeMode="tail"
      selectable={true}
      style={
        Array [
          Object {
            "color": "#A3A5A9",
            "fontSize": 16,
            "fontWeight": "900",
          },
          Object {
            "fontFamily": "DejaVuSans",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
